server/storage/database_test.go

```diff
@@ -501,3 +482,133 @@ func TestDBGetChecksumNotFound(t *testing.T) {
 	require.Error(t, err)
 	require.IsType(t, ErrNotFound{}, err)
 }
+
+func TestGetVersions(t *testing.T) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	_, store := SetUpSQLite(t, tempBaseDir)
+	defer os.RemoveAll(tempBaseDir)
+
+	_, repo, _, err := testutils.EmptyRepo("gun")
+	require.NoError(t, err)
+
+	ts, err := repo.SignTimestamp(data.DefaultExpires("timestamp"))
+	tsJSON1, err := json.Marshal(ts)
+	require.NoError(t, err)
+	store.UpdateCurrent(
+		"gun",
+		MetaUpdate{Role: "timestamp", Version: 1, Data: tsJSON1},
+	)
+
+	ts, err = repo.SignTimestamp(data.DefaultExpires("timestamp"))
+	tsJSON2, err := json.Marshal(ts)
+	require.NoError(t, err)
+	store.UpdateCurrent(
+		"gun",
+		MetaUpdate{Role: "timestamp", Version: 2, Data: tsJSON2},
+	)
+	checksumBytes := sha256.Sum256(tsJSON2)
+	checksum := hex.EncodeToString(checksumBytes[:])
+
+	ts, err = repo.SignTimestamp(data.DefaultExpires("timestamp"))
+	tsJSON3, err := json.Marshal(ts)
+	require.NoError(t, err)
+	store.UpdateCurrent(
+		"gun",
+		MetaUpdate{Role: "timestamp", Version: 3, Data: tsJSON3},
+	)
+
+	versions, err := store.GetVersions("gun", "timestamp", "", 0)
+	require.NoError(t, err)
+
+	require.Len(t, versions, 3)
+	require.EqualValues(t, tsJSON3, versions[0])
+	require.EqualValues(t, tsJSON2, versions[1])
+	require.EqualValues(t, tsJSON1, versions[2])
+
+	versions, err = store.GetVersions("gun", "timestamp", checksum, 1)
+	require.NoError(t, err)
+
+	require.Len(t, versions, 1)
+	require.EqualValues(t, tsJSON1, versions[0])
```

Comment:
Would it be possible to add a couple more checks here?  Some of these may be redundant (so please feel free to reject any or all):

1. `store.GetVersions("gun", "timestamp", "", 1)` should return just `versions[2]` (I think?)
1. `store.GetVersions("gun", "timestamp", checksum, 0)` should return `versions[0]`
1. `store.GetVersions("gun", "timestamp", checksum, 80)` should also return `versions[0]`
1. `store.GetVersions("gun", "timestamp", "", 80)` should also return all?

The number being > than what's there case is the particular one I wanted to test.